### ISSUE TITLE
IE rotation collision bug 

### ISSUE THREAD

**Issue by Antbrooksuk** [craftyjs/Crafty#175]
Rotating a sprite causes collisions to fail in ie6/7/8

[Google Groups discussion](https://groups.google.com/forum/?fromgroups=#!topic/craftyjs/Lee2AdwwwG8)

**Comment by sorenbs** [comment]
The rotation system needs to be looked at in general.

This simple example behaves strange in all browsers http://jsfiddle.net/qpEGp/

**Comment by Antbrooksuk** [comment]
I've removed rotation from my game for the moment, it would be good to have
a dirty fix or workaround if possible? :)

On Mon, Jan 16, 2012 at 10:05 PM, sorenbs <
reply@reply.github.com

**Comment by austinh** [comment]
I am still having problems with this issue in IE. :( Not sure what to do. Would love it if someone could at least give a dirty fix or lead me on the right track.

**Comment by starwed** [comment]
Could someone who experiences this explain the issue?  The jsfiddle @sorenbs  linked to doesn't load right for me, and the google groups discussion is a bit vague.  (The last links that Anthony Brooks gave seem to be dead?)

**Comment by sbyrnes321** [comment]
I was looking into why that jsfiddle wasn't working. It seems that the collision component makes the draggable component not work! Is this a known issue? Simple case: http://jsfiddle.net/jLzJm/ On my firefox, you can only drag one of the two boxes.

**Comment by starwed** [comment]
It looks like adding "Collision" is blocking other mouse events as well; Maybe the added collision polygon is getting them? ( [jsfiddle](http://jsfiddle.net/jLzJm/3/) )

  I'll open a new issue to investigate, since this seems separate from the rotation/collision issue.

**Comment by sbyrnes321** [comment]
As discussed in #448, the collision box in the above examples is not reset (with .collision()) after the width and height of the component are set to nonzero values.

After taking that into account, here is code that works in firefox and IE10 with the latest crafty. It also works with the last crafty release 0.5.3 (so this is not revealing a problem fixed by #450):

```
Crafty.init(500,500);
Crafty.canvas.init();
var count = 0;
Crafty.e( "2D, Draggable, DOM, Color, Collision" )
    .attr( { w: 20, h: 300 } )
    .collision()
    .color( "red" )
    .attr( { rotation: 30 } );
Crafty.e( "2D, Draggable, DOM, Color, Collision" )
    .attr( { w: 20, h: 300 } )
    .collision()
    .color( "green" )
    .onHit( "Color", function () { count++; });
Crafty.e("2D, DOM, Text")
    .bind("EnterFrame", function () {this.text(count);});
```

It also works with IE10 in IE9 mode, but does NOT work with IE10 in IE8 mode. In fact the screen is blank and no errors appear on the console either. But that fact has nothing to do with the rotation. (Deleting the "rotation" line doesn't help.) It also has nothing to do with recent changes: The screen is blank in the 0.5.3 release, and is blank with the latest development build from today. Instead, this is probably related to the more basic problems pointed out in #456.

So to sum up, I very strongly suspect that this issue can be closed, because whatever problems exist, they are probably covered by #448, #450, and #456. (Unless, of course, there is a difference between IE8 and "IE10 in IE8 mode", which I doubt, but I don't have IE8 to check.)

**Comment by starwed** [comment]
Ok, closing.
